unblob/extractor.py

```diff
@@ -36,14 +37,52 @@ def carve_chunk_to_file(carve_path: Path, file: io.BufferedIOBase, chunk: Chunk)
             f.write(data)
 
 
-def fix_permissions(outdir: Path):
+def fix_permission(path: Path):
+    if path.is_file():
+        path.chmod(0o644)
+    elif path.is_dir():
+        path.chmod(0o775)
+
+
+def is_safe_path(basedir: Path, path: Path) -> bool:
+    try:
+        basedir.joinpath(path).resolve().relative_to(basedir.resolve())
+    except ValueError:
+        return False
+    return True
+
+
+def fix_symlink(path: Path, outdir: Path) -> Path:
+    """Fix symlinks by rewriting absolute symlinks to make them point within
+    the extraction directory (outdir), if it's not a relative symlink it is
+    either removed it it attempts to traverse outside of the extraction directory
+    or rewritten to be fully portable (no mention of the extraction directory
+    in the link value)."""
+    target = Path(os.readlink(path))
+
+    if target.is_absolute():
+        target = Path(target.as_posix().lstrip("/"))
+    else:
+        target = path.resolve()
+
+    safe = is_safe_path(outdir, target)
+
+    if not safe:
+        logger.warning(f"Path traversal attempt through symlink, removing {target}.")
```

Comment:
I would at least give an error here, nobody cares about warnings usually

Reply:
Makes sense. Moved to error.